source/common/network/connection_impl.h

```diff
@@ -51,10 +51,14 @@ class ConnectionImpl : public virtual Connection,
                        public BufferSource,
                        public TransportSocketCallbacks,
                        protected Logger::Loggable<Logger::Id::connection> {
-public:
+protected:
```

Comment:
nit: unless it can't be done, please move so the class is defined from public -> protected -> private. Same below.

Reply:
Should be doable.